blog/content/post/2020-04-03-trenchboot-nlnet-lz-validation.md

```diff
@@ -0,0 +1,887 @@
+    [    0.000000] x86/fpu: Supporting XSAVE feature 0x001: 'x87 floating point registers'
+
+    (...)
+
+    <<< Welcome to NixOS 20.09.git.a070e686875 (x86_64) - ttyS0 >>>
+
+    Run 'nixos-help' for the NixOS manual.
+    ```
+
+**Verification**: As expected, bootloader executes Linux kernel directly.
+Platform booted without DRTM then.
+
+2. Reboot platform once again. In GRUB menu choose `"NixOS - Secure Launch"`
+entry.
+
+Once again, collect logs during boot to be able to verify them. Using `dmesg`
+command in NixOS doesn't work as in previous case. Correct bootlog is shown
+below.
+
+    ```
+    grub_cmd_slaunch:122: check for manufacturer
+    grub_cmd_slaunch:126: check for cpuid
+    grub_cmd_slaunch:136: set slaunch
+    grub_cmd_slaunch_module:156: check argc
+    grub_cmd_slaunch_module:161: check relocator
+    grub_cmd_slaunch_module:170: open file
+    grub_cmd_slaunch_module:175: get size
+    grub_cmd_slaunch_module:180: allocate memory
+    grub_cmd_slaunch_module:192: addr: 0x100000
+    grub_cmd_slaunch_module:194: target: 0x100000
+    grub_cmd_slaunch_module:196: add module
+    grub_cmd_slaunch_module:205: read file
+    grub_cmd_slaunch_module:215: close file
+    grub_slaunch_boot_skinit:41: real_mode_target: 0x8a000
+    grub_slaunch_boot_skinit:42: prot_mode_target: 0x1000000
+    grub_slaunch_boot_skinit:43: params: 0xcfe7745early console in extract_kernel
+    input_data: 0x00000000023eb3b1
+    input_len: 0x0000000000424e94
+    output: 0x0000000001000000
+    output_len: 0x00000000017e7398
+    kernel_total_size: 0x000000000142c000
+    trampoline_32bit: 0x000000000009d000
+    booted via startup_32()
+    Physical KASLR using RDTSC...
+    Virtual KASLR using RDTSC...
+
+    Decompressing Linux... Parsing ELF... Performing relocations... done.
+    Booting the kernel.
+    [    0.000000] Linux version 5.1.0 (nixbld@localhost) (gcc version 9.2.0 (GCC)) #1-NixOS SMP Thu Jan 1 00:00:01 UTC 1970
+    [    0.000000] Command line: BOOT_IMAGE=(hd0,msdos1)/nix/store/ymvcgas7b1bv76n35r19g4p142v4cr0b-linux-5.1.0/bzImage systemConfig=/nix/store/j4
+    [    0.000000] x86/fpu: Supporting XSAVE feature 0x001: 'x87 floating point registers'
+
+    (...)
+
+    <<< Welcome to NixOS 20.09.git.a070e686875 (x86_64) - ttyS0 >>>
+
+    Run 'nixos-help' for the NixOS manual.
+    ```
+
+**VERIFICATION**: As expected, before Linux kernel, there should be `slaunch`
+module executed. It proves that DRTM is enabled. There is no information about
+LZ execution because it is non-debug version.
+
+### Landing Zone
+
+Actually, there are few aspects which can be verified in LZ. We will focus on
+those two:
+- check if LZ utilizes SHA256 algorithm when using TPM2.0 module
+- check if LZ debug option can be enabled
+
+##### check if LZ utilizes SHA256 algorithm when using TPM2.0 module
+
+1. Reboot platform and boot to NixOS via `"NixOS - Secure Launch"` entry in GRUB
```

Comment:
Assuming user already booted DRTM version a reboot here is not required

Reply:
I don't want to assume this. Let's suppose user wants to go through LZ verification only and not previous processes. In this case, it is safer to leave that.

Reply:
@piotr-kleins then maybe change it to `Reboot (...) if not done already`?

Reply:
Reworded